src/fn/edit-contacts.js

```diff
@@ -0,0 +1,169 @@
+const minimist = require('minimist');
+const environment = require('../lib/environment');
+const fs = require('../lib/sync-fs');
+const path = require('path');
+const { warn, info } = require('../lib/log');
+const pouch = require('../lib/db');
+const toDocs = require('./csv-to-docs');
+
+module.exports = ()=> {
+  const args = parseExtraArgs(environment.pathToProject, environment.extraArgs);
+  const db = pouch();
+  const docDirectoryPath = args.docDirectoryPath;
+  fs.mkdir(docDirectoryPath);
+  const saveJsonDoc = doc => fs.write(`${docDirectoryPath}/${doc._id}.doc.json`, toDocs.toSafeJson(doc) + '\n');
+
+  const csvDir = `${environment.pathToProject}/csv`;
+  if(!fs.exists(csvDir)) {
+    warn(`No csv directory found at ${csvDir}.`);
+    return Promise.resolve();
+  }
+
+  return fs.recurseFiles(csvDir)
+  .filter(name => name.endsWith('.csv'))
+  .reduce((promiseChain, csv) =>
+    promiseChain
+    .then(() => {
+      info('Processing CSV file:', csv, '…');
+
+
+      const nameParts = fs.path.basename(csv).split('.');
+      const prefix = nameParts[0];
+      info('Processing CSV file:', prefix, '…');
+
+      switch(prefix) {
+        case 'contact':  return downloadDocs(csv, getIDs(csv),db, args);
+        default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
+      }
+    })
+    .then(docs => addToModel(csv, docs)),
+    Promise.resolve())
+
+  .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
+  .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+};
+
+const model = {
+  csvFiles: {},
+  docs: {},
+  exclusions: [],
+  users: []
+};
+
+const addToModel = (csvFile, docs) => {
+    csvFile = csvFile.match(/^(?:.*[\/\\])?csv[\/\\](.*)\.csv$/)[1]; // eslint-disable-line no-useless-escape
+    model.csvFiles[csvFile] = docs;
+    docs.forEach(doc => {
+      model.docs[doc._id] = doc;
+    });
+  };
+
+function getIDs(csv) {
+    const { rows, cols } = fs.readCsv(csv);
+    var index = cols.indexOf('uuid');
+    if (index === -1){
+     throw Error('missing "uuid" column.');
+   }
+   return rows.map((item) => item[index]);
+}
+
+function processContacts(contactType, csv, ids, contactDocs, args){
+  const { rows, cols } = fs.readCsv(csv);
+  var colNames = args.colNames;
+  var index = cols.indexOf('uuid');
+  var toIncludeIndex = [];
+  if (colNames.length === 0) {
+    warn(' No columns specified, the script will add all the columns in the CSV!');
+    colNames = cols;
+  } else {
+    var splitFileColumns = [];
+    var fileColumn;
+    for(fileColumn of cols) {
+      if (fileColumn !== null) {
+        splitFileColumns.push(fileColumn.split(':')[0]);
+      }      
+    }
+    const columnsValid = colNames.every(function(column) { return splitFileColumns.includes(column);});
+    if (!columnsValid){
+      throw Error('The column name(s) specified do not exist.');
+    }
```

Comment:
As far as I can tell, this part only evaluates whether every `args.colNames` exists as a column in the csv.
Consider pulling this out as a separate function that returns a Boolean that represents whether the `args.colNames` are valid.

Reply:
Done